graph/graphtest/graphtest.go

```diff
@@ -513,8 +512,15 @@ func TestLoadTypedQuads(t testing.TB, gen DatabaseFunc, conf *Config) {
 					pq = quad.Time(time.Unix(seconds, nanos).UTC())
 				}
 			}
+			// Right now it looks like only quad.Time has an Equaler interface.
 			if eq, ok := pq.(quad.Equaler); ok {
 				assert.True(t, eq.Equal(got), "Failed to roundtrip %q (%T), got %q (%T)", pq, pq, got, got)
+				
+				// Built-in time.Equal does not compare timezone values.
+				// Note that asserts.Equal is not being used here as it is causing a panic in the LevelDB backend.
+				if _, ok := pq.(quad.Time); ok {
```

Comment:
Well, `quad.Equaler` was created to avoid these direct type assertions, so we should track down the bug in LevelDB instead.
Regarding time zones, we don't store them for now. Time is saved as seconds+nanos in UTC, and are returned in a local timezone (if I recall correctly).

I was trying to say that we can add additional checks at the end of the test that ensures, that it can round-trip values/quads second time.

Bug is the following:
1) Quad is added with some timezone, hash was made from this value on quad insertion.
2) Iterator and QS can still access this time value either from quad/value iterator, or by calling `qs.ValueOf` with original time.
3) Quad returned by QS contains correct hash, but time value is reconstructed in different time zone. Thus, if we try to use this new time value from quad instead of original one, it will fail, since hash is different now.

The fix might be to use UTC for hash calculation. But first you'll need to fix this test :)